common/src/main/kotlin/org/jetbrains/kotlinconf/api/KotlinConfApi.kt

```diff
@@ -11,7 +11,8 @@ import org.jetbrains.kotlinconf.data.*
 
 internal expect val END_POINT: String
 
-class KotlinConfApi(private val userId: String) {
```

Comment:
Please consider keeping single instance creation and user state info here.
It's quite complex to manage requests and state with multiple instances.

Reply:
Not sure what do you mean? Right now user can use app without user id (which is code) and after some time provide it. With this logic, we cannot keep the way it worked.

Reply:
`KotlinConfApi` is stateless. Could we convert it to an object?

Reply:
We can, but it would give us nothing. It is used only in `KotlinConfDataRepository` which also have only a single instance anyway. The only change from making `KotlinConfApi` to object declaration is that it would be additionally held in a static field what are a bit challenging for Garbage Collector. Also, I am not sure how about Kotlin/Native and object declarations.